src/interfaces.ts

```diff
@@ -13,13 +13,28 @@ export interface IGitHubRepoLabels {
   }
 }
 
+export enum MergeStateStatus {
```

Comment:
This enum is exposed in the GitHub API and AFAIK its experimental and can be broken at any time

can you please provide documentation showing its production ready?

 That's my current need. Stability.

Reply:
Secondly the mergeable status does not report the "merge chances" that are a business requirement for the system we are using.

Reply:
What do you mean exactly with the "merge chances"? From the readme I didn't understand what was the goal exactly. Can you provide an example please?

That property has been in the REST API since a lot of time (https://docs.github.com/en/rest/reference/pulls#get-a-pull-request, mergeable_state property) it's been in preview in the GraphQL API only.

The motivation for these changes are for us at Jellyfin project: we've been some time wondering why PRs without conflicts were not getting labeled and I discovered that it was indeed the mschilde's action misstep. Seeing that the repo is inactive, I made the PR in both sides :)

Reply:
Yes this field is list but it's not the recommendation.

https://docs.github.com/en/rest/guides/getting-started-with-the-git-database-api#checking-mergeability-of-pull-requests

https://docs.github.com/en/rest/reference/pulls#get-a-pull-request says:

> You can review the status of the test commit using the mergeable key

https://github.community/t/pullrequest-mergeable-state-possible-values/13926/2 From 2019 says it's should be considered *unofficial*. Though reading it again knowing it in the GraphQL maybe it's official.... 🤯 It was there a few weeks ago but it seems to be a recent addition this year https://github.community/t/pullrequest-mergeable-state-possible-values/13926/4

***

It sounds like you need something very similar to us, but I am not sure how much testing you've done with this enum... it's really not what you want.

You can see my testing here https://github.com/prince-chrismc/conan-center-index/pulls?q=is%3Apr+is%3Aclosed and Even more again the upstream project

➡️ Let's us this fake PR to test with.... https://github.com/prince-chrismc/conan-center-index/pull/24


***

> What do you mean exactly with the "merge chances"?

I struggled with defining this so if you have any suggestions for improvements I am all ears!

Let me give you an example.

Reply:
Damn. I love when I am wrong!

Trying to write an explanations why that field should not be used... I think I convinced myself it might actually work and it would reduce some of the complexity!

I have this PR which needs a rebase but even my code would not detect the problem.

![image](https://user-images.githubusercontent.com/16867443/112637324-67638e80-8e14-11eb-9e45-7cef5f37da0c.png)


Checking a few PRs, none of them are marked "dirty" so we need to figure out the exact conditions to get that to trigger

Reply:
I think there's something to improve here but might need a tweak

![image](https://user-images.githubusercontent.com/16867443/112638478-ad6d2200-8e15-11eb-9fe7-16956cf6a80a.png)

Reply:
@prince-chrismc If I'm understanding your issue correctly, what you want is to enforce in GitHub to always have updated branches with rebases on PRs (we already have that), so they become ``BEHIND`` (this repo itself seems to enforce that). That way, already merged commits will never go in **if you're using a rebase strategy**. This basically means you want the ``mergeable_only`` boolean that I added, so the label is removed as soon as everything is clean, checks are passed and reviews are done.

> I struggled with defining this so if you have any suggestions for improvements I am all ears!

What do you think of my definition? I think it's more clear.

> From 2019 says it's should be considered unofficial.

I've been playing with GH API since some time and, in my experience, the field it's very reliable. I understand your stability concerns but, given the recent advancements in the documentation, I think that getting it in the GraphQL is just a matter of time.

The REST API one is stable afaik. When in the docs they mean to use the ``mergeable`` key, it's true, but it doesn't provide the ``ful picture``: You can have a non-mergeable PR (based on your CI rules, for instance) but you can bypass them (if you, for instance, don't let the collaborators of your project to merge if the requirements doesn't pass). So that PR it's mergeable, although the API doesn't say that.

Reply:
@prince-chrismc Yeah, so basically, if I'm understanding everyhting correctly, I think that you want the label to be removed only when both ``mergeable == MERGEABLE`` and ``mergeableStateStatus === CLEAN`` (that's the ``mergeable_only`` flag) and we want it as soon as ``mergeableStateStatus !== DIRTY``, right?

Reply:
This is new to me, so I am trying to go through it and make sure I understand the nuances and caveats.